### ISSUE TITLE
array-callback-return: diagnostic anchored on inner `switch` instead of the array callback, breaking eslint-disable-next-line suppression

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25632]
## Summary

The native `array-callback-return` rule reports its diagnostic on the inner `switch` statement rather than on the array-method callback where ESLint reports it. A `// eslint-disable-next-line array-callback-return` directive placed directly above the callback — the correct, ESLint-compatible position — therefore does not suppress the finding, and with `--report-unused-disable-directives` the genuinely-needed directive is additionally flagged as unused. In a large migration this turns valid suppression directives into spurious unused-directive errors.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.js`:

```js
const parts = [{type: 'a', value: 1}]

const out = parts
	// eslint-disable-next-line array-callback-return
	.map(p => {
		switch (p.type) {
			case 'a':
				return p.value
		}
	})
	.join(' ')
```

`.oxlintrc.json`:

```json
{
	"plugins": [],
	"categories": {"correctness": "off"},
	"rules": {
		"array-callback-return": ["error", {"allowImplicit": true, "checkForEach": false}]
	}
}
```

Run:

```
oxlint -c .oxlintrc.json --report-unused-disable-directives repro.js
```

## Expected

ESLint anchors the violation on the **callback** (the `p => {` arrow function — `4:9` on a directive-free copy of the fixture), so the `eslint-disable-next-line` directive on the line directly above the callback suppresses it:

- ESLint 7.23.0 (core rule, `--report-unused-disable-directives`): no output, exit 0 — directive stays live.
- ESLint 9.39.4 (flat config, core rule, `--report-unused-disable-directives`): no output, exit 0 — directive stays live.

## Actual

oxlint 1.78.0 emits:

```
repro.js:6:3: error eslint(array-callback-return): Callback for array method "Array.prototype.map" may fall through a `switch` without returning help: This `switch` has no `default` case, so the callback may reach the end without a `return`. Add a `default` that returns/throws, or add a final `return` after the `switch`.
repro.js:4:2: warning: Unused eslint-disable directive (no problems were reported).
```

The diagnostic is anchored on the `switch` (line 6) instead of the `.map(...)` callback (line 5). The directive on line 4 scopes to line 5 — matching ESLint — so it does not cover oxlint's line-6 anchor, and oxlint both errors on the violation and wrongly flags the directive as unused. (On 1.69.0, running the same rule through the bundled real-ESLint jsPlugin `eslint-js/array-callback-return` anchored on the callback line, matching ESLint — so the correct location is reachable; the native Rust rule diverges.)

## Cross-checked

| Engine | Version | Anchor (no directive) | Directive above callback |
| --- | --- | --- | --- |
| ESLint (core rule) | 7.23.0 | `4:9` — callback | suppressed, exit 0 |
| ESLint (core rule, flat config) | 9.39.4 | `4:9` — callback | suppressed, exit 0 |
| oxlint (native rule) | 1.78.0 | `5:3` — `switch` | not suppressed + unused-directive warning, exit 1 |
